### DISCUSSION TITLE
Lapse responses and response times

### DISCUSSION THREAD

**Discussion by jCalderTravis** [ampl-psych/EMC2#77] [General]
Thanks for a very interesting looking package, and for the associated documentation.

From the preprint and documentation I didn't find anything about accounting for lapse trials / contaminant processes. From my understanding there is good evidence that a small proportion of responses and response times are not due to the normal decision process but are the result of lapses. Furthermore, it is known that such lapse responses and response times can have a substantial bias effect on modelling fits unless such contaminant processes are to some extent explicitly modelled (e.g. Ratcliff & Tuerlinckx, 2002, https://doi.org/10.3758/BF03196302).

Is there currently any way to fit a lapse rate parameter in EMC2 (describing the proportion of trials for which the response and RT is random), or are there any plans to include a lapse rate parameter? 

Thank you for your time.

**Comment by frankhezemans** [comment]
At the moment, none of the models explicitly incorporate a lapse rate parameter. However, the trial-wise log likelihood is bounded below with an arbitrary small value, `min_ll` (default `log(1e-10)`, see e.g. [here](https://github.com/ampl-psych/EMC2/blob/ea4c1f50b537df1af5ca83a30d94344ab7777892/R/model_DDM.R#L132-L134)). Informally, you can think of this as fitting the model with a _fixed_ lapse rate of `min_ll`, in the sense that this `min_ll` value would be assigned to any trials with implausibly fast or slow response times (i.e., those generated from the contaminant process).

**Reply by jCalderTravis** [reply]
Thanks for your reply, and for the details. Having a lower bound on the log likelihood is certainly helpful. I think a more systematic treatment of lapses would still be an enhancement to an already great package, because there is evidence that a substantial proportion of trials come from contaminant processes. Let me know if you think it would be worthwhile for me to raise this as a feature request.

**Reply by frankhezemans** [reply]
Thanks for your continued interest. I would say, by all means, please go ahead and raise an issue for this feature request. However, I suspect this won’t be a very high priority, as development efforts are currently focused on models of stop signal paradigms and across-trial learning dynamics, among other things.

I remember asking @andrewheathcote about contaminant processes / lapse rate a while ago, and he raised concerns about the identifiability of the lapse rate parameter in realistic empirical datasets. Also, there are some theoretical critiques of the contaminant mixture approach. Quoting from [Hawkins et al. (2019)](https://doi.org/10.1016/j.cogpsych.2019.05.002):

> Although the addition of a contaminant mixture distribution stabilizes parameter estimates (Ratcliﬀ & Tuerlinckx, 2002), it provides no insight into _why_ some responses were not generated by the process of evidence accumulation. Furthermore, since it assumes that an observed choice-RT pair is generated from the evidence accumulation process _or_ the contaminant distribution, it must assume that there is a proportion of trials in which the participant never initiated the evidence accumulation process, regardless of the duration of the observed RT. This begs the question of why this occurs.

**Reply by jCalderTravis** [reply]
Sorry for the very slow reply, and thank you very much for your response. 

To me the lapse rate make sense as a way to catch all those trials where the normal mechanism is not taking place, e.g. participant gets distracted by a noise outside, or sneezes. 

I will look into whether there is evidence that a lapse rate parameter would be identifiable before raising raising a feature request.

Thanks again!
